commons-core/src/main/scala/com/avsystem/commons/misc/Timestamp.scala

```diff
@@ -0,0 +1,48 @@
+package com.avsystem.commons
+package misc
+
+import com.avsystem.commons.serialization.{GenCodec, GenKeyCodec, IsoInstant}
+
+import scala.concurrent.duration.{FiniteDuration, TimeUnit}
+
+/**
+  * Millisecond-precision, general purpose, cross compiled timestamp representation.
+  *
+  * @param millis milliseconds since UNIX epoch, UTC
+  */
+class Timestamp(val millis: Long) extends AnyVal {
```

Comment:
I wanted to have a sensible cross-compiled, immutable, comparable, serializable and human-readable type that could be put into RPC interfaces.

I know that in order to show it to the user, fronted will have to convert it to something raw (millis, `js.Date`) so it can be passed to `moment.js` or whatever other formatting library. However, this can now happen at the edge of our codebase and `JDate` will no longer pollute it, most importantly our RPC interfaces.